config/Dockerfiles/cloud-image-compose/virt-customize.sh

```diff
@@ -47,33 +46,23 @@ function clean_up {
   ln -sf $(find . -name "*.qcow2") test_subject.qcow2
   popd
   kill $(jobs -p)
-  # In case of first build, ensure image_artifacts dir exists
-  mkdir -p ${image_artifacts}
-  cp -rp ${CURRENTDIR}/images/* ${CURRENTDIR}/logs ${image_artifacts}
-  #ln -sf ${ARTIFACT_MNT}/${package}  ${ARTIFACT_MNT}/latest
 }
 trap clean_up EXIT SIGHUP SIGINT SIGTERM
 
 { #group for tee
 
 mkdir -p ${CURRENTDIR}/images
 
+RPM_LIST=""
+REPO_LIST="--repofrompath=${package},file:///etc/yum.repos.d"
 # Add custom rpms to image
-while read rpm ; do
-    rpm_name=$(echo $rpm | cut -d ' ' -f 1)
-    TIMEDIFF=$(expr $(date '+%s') - $(date '+%s' -d $(echo $rpm | cut -d ' ' -f 2)))
-    # Only rpms saved within last day
-    if [[ $TIMEDIFF -le 86400 ]]; then
-        virt-copy-in -a ${DOWNLOADED_IMAGE_LOCATION} ${rpm_repo}/${rpm_name} /etc/yum.repos.d/
-        for pkg in $(repoquery --disablerepo=\* --enablerepo=${rpm_name} --repofrompath=${rpm_name},${rpm_repo} --all | grep -v '\-debug\|\-devel' | rev | cut -d '-' -f 3- | rev ) ; do
-            if ! virt-customize -a ${DOWNLOADED_IMAGE_LOCATION} --run-command "yum install -y --nogpgcheck --repofrompath=testrepo,file:///etc/yum.repos.d/${rpm_name} ${pkg}" ; then
-                if [ $package == $rpm_name ] ; then
-                    sed -i "/^${package}/d" ${manifest}
-                fi
-                exit 1
-            fi
-        done
-    fi
-done < $manifest
+virt-copy-in -a ${DOWNLOADED_IMAGE_LOCATION} ${rpm_repo}/*.rpm ${rpm_repo}/repodata /etc/yum.repos.d/
+
+for pkg in $(dnf repoquery --disablerepo=\* --enablerepo=${package} --repofrompath=${package},${rpm_repo} --all | egrep -v '\-debug\|\-devel|.src' | rev | cut -d '-' -f 3- | rev ) ; do
+    RPM_LIST="${RPM_LIST} ${pkg}"
+done
+if ! virt-customize -a ${DOWNLOADED_IMAGE_LOCATION} --run-command "dnf install -y --nogpgcheck ${REPO_LIST} ${RPM_LIST}" ; then
+    echo "failure"
```

Comment:
I think we need an actual exit != 0 here, unless the exit code is already nonzero because of the if failing?

Reply:
yes, good catch.